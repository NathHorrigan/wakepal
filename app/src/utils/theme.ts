// Utility functions to access theming
export const colors = {
  coral: '#F86A6A',
  cyan: '#0E7C86',
  paleGreen: '#A3D9A5',
  darkBlue: '#2680C2',
}

export const fonts = {
  medium: 'SFProRounded-Medium',
  semiBold: 'SFProRounded-SemiBold',
  bold: 'SFProRounded-Bold',
}
