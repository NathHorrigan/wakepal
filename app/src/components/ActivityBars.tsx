import React, { SFC } from 'react'
import styled from 'styled-components/native'
import { colors, fonts } from '@utils/theme'

interface ActivityBarProps {
  label: string
  value: number
  goal: number
  color: string
  unit?: string
}

interface ActivityBarsProps {
  activities: ActivityBarProps[]
}

export const ActivityBar: React.SFC<ActivityBarProps> = ({
  value,
  goal,
  color,
}) => {
  const width = Math.min(100, (value / goal) * 100)
  return (
    <Activity>
      <ActivityBarOuter>
        <ActivityBarInner color={color} value={width} />
      </ActivityBarOuter>
    </Activity>
  )
}

export const ActivityBars: React.SFC<ActivityBarsProps> = ({ activities }) => {
  return (
    <BarsContainer>
      <LabelColumn>
        {activities.map(({ label }) => (
          <ActivityName key={`${label}-label`}>{label}</ActivityName>
        ))}
      </LabelColumn>
      <BarsColumn>
        {activities.map(activity => (
          <ActivityBar key={`${activity.label}-bar`} {...activity} />
        ))}
      </BarsColumn>
      <StatsColumn>
        {activities.map(({ label, value, goal, unit }) => (
          <ActivityStats key={`${label}-stat`}>
            {value}/{goal} {unit}
          </ActivityStats>
        ))}
      </StatsColumn>
    </BarsContainer>
  )
}

const BarsContainer = styled.View`
  width: 90%;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: flex-start;
`

const Activity = styled.View`
  height: 10px;
  margin-bottom: 8px;
`

const LabelColumn = styled.View`
  display: flex;
  justify-content: flex-start;
  margin-top: 2px;
`

const BarsColumn = styled.View`
  display: flex;
  flex-grow: 1;
  justify-content: flex-start;
  margin-top: 8px;
`

const StatsColumn = styled.View`
  display: flex;
  margin-top: 5px;
  justify-content: flex-start;
`

const ActivityName = styled.Text`
  color: #4e4e4e;
  font-size: 12px;
  font-family: ${fonts.bold};
  margin-right: 15px;
  margin-bottom: 4px;
`

const ActivityStats = styled.Text`
  color: #9e9e9e;
  font-size: 10px;
  font-family: ${fonts.semiBold};
  margin-left: 15px;
  height: 15px;
  margin-bottom: 3px;
`

const ActivityBarOuter = styled.View`
  height: 10px;
  flex-grow: 1;
  background: ${colors.lightGrey};
  border-radius: 10px;
`

const ActivityBarInner = styled.View`
  height: 10px;
  width: ${props => props.value}%;
  background: ${props => props.color};
  border-radius: 10px;
`

export default ActivityBars
