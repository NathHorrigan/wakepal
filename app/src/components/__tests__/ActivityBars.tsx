import React from 'react'
import renderer from 'react-test-renderer'

import ActivityBars from '../ActivityBars'
import { colors } from '@utils/theme'

it('renders correctly', () => {
  // Check button matches screenshot
  const tree = renderer
    .create(
      <ActivityBars
        activities={[
          { label: 'Steps', value: 8768, goal: 10000, color: colors.coral },
          { label: 'Floors', value: 16, goal: 25, color: colors.cyan },
          {
            label: 'Calories',
            value: 1000,
            goal: 1200,
            color: colors.paleGreen,
          },
          {
            label: 'Water',
            value: 4,
            goal: 6,
            color: colors.paleBlue,
            unit: 'Pints',
          },
        ]}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
