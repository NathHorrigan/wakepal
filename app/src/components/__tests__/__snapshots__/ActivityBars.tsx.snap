// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "width": "90%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "justifyContent": "flex-start",
          "marginTop": 2,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#4e4e4e",
            "fontFamily": "SFProRounded-Bold",
            "fontSize": 12,
            "marginBottom": 4,
            "marginRight": 15,
          },
        ]
      }
    >
      Steps
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#4e4e4e",
            "fontFamily": "SFProRounded-Bold",
            "fontSize": 12,
            "marginBottom": 4,
            "marginRight": 15,
          },
        ]
      }
    >
      Floors
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#4e4e4e",
            "fontFamily": "SFProRounded-Bold",
            "fontSize": 12,
            "marginBottom": 4,
            "marginRight": 15,
          },
        ]
      }
    >
      Calories
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#4e4e4e",
            "fontFamily": "SFProRounded-Bold",
            "fontSize": 12,
            "marginBottom": 4,
            "marginRight": 15,
          },
        ]
      }
    >
      Water
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "flexGrow": 1,
          "justifyContent": "flex-start",
          "marginTop": 8,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 10,
            "marginBottom": 8,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#E1E1E1",
              "borderRadius": 10,
              "flexGrow": 1,
              "height": 10,
            },
          ]
        }
      >
        <View
          color="#F86A6A"
          style={
            Array [
              Object {
                "backgroundColor": "#F86A6A",
                "borderRadius": 10,
                "height": 10,
                "width": "87.68%",
              },
            ]
          }
          value={87.68}
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "height": 10,
            "marginBottom": 8,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#E1E1E1",
              "borderRadius": 10,
              "flexGrow": 1,
              "height": 10,
            },
          ]
        }
      >
        <View
          color="#0E7C86"
          style={
            Array [
              Object {
                "backgroundColor": "#0E7C86",
                "borderRadius": 10,
                "height": 10,
                "width": "64%",
              },
            ]
          }
          value={64}
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "height": 10,
            "marginBottom": 8,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#E1E1E1",
              "borderRadius": 10,
              "flexGrow": 1,
              "height": 10,
            },
          ]
        }
      >
        <View
          color="#A3D9A5"
          style={
            Array [
              Object {
                "backgroundColor": "#A3D9A5",
                "borderRadius": 10,
                "height": 10,
                "width": "83.33333333333334%",
              },
            ]
          }
          value={83.33333333333334}
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "height": 10,
            "marginBottom": 8,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#E1E1E1",
              "borderRadius": 10,
              "flexGrow": 1,
              "height": 10,
            },
          ]
        }
      >
        <View
          color="#62B0E8"
          style={
            Array [
              Object {
                "backgroundColor": "#62B0E8",
                "borderRadius": 10,
                "height": 10,
                "width": "66.66666666666666%",
              },
            ]
          }
          value={66.66666666666666}
        />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "justifyContent": "flex-start",
          "marginTop": 5,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#9e9e9e",
            "fontFamily": "SFProRounded-SemiBold",
            "fontSize": 10,
            "height": 15,
            "marginBottom": 3,
            "marginLeft": 15,
          },
        ]
      }
    >
      8768
      /
      10000
       
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#9e9e9e",
            "fontFamily": "SFProRounded-SemiBold",
            "fontSize": 10,
            "height": 15,
            "marginBottom": 3,
            "marginLeft": 15,
          },
        ]
      }
    >
      16
      /
      25
       
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#9e9e9e",
            "fontFamily": "SFProRounded-SemiBold",
            "fontSize": 10,
            "height": 15,
            "marginBottom": 3,
            "marginLeft": 15,
          },
        ]
      }
    >
      1000
      /
      1200
       
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#9e9e9e",
            "fontFamily": "SFProRounded-SemiBold",
            "fontSize": 10,
            "height": 15,
            "marginBottom": 3,
            "marginLeft": 15,
          },
        ]
      }
    >
      4
      /
      6
       
      Pints
    </Text>
  </View>
</View>
`;
